src/Analysis/Engine/Test/LanguageServerTests.cs

```diff
@@ -992,6 +992,19 @@ class C:
             );
         }
 
+        [TestMethod, Priority(0)]
+        public async Task OnTypeFormatting() {
+            var s = await CreateServer();
+
+            var mod = await AddModule(s, "def foo  ( ) :\n    x = a + b\n    x+= 1");
```

Comment:
One of the good rules about the test is that the person who sees the test for the first time (but familiar with the code base) should be able to understand what's going on. 

Here, without digging into implementation, I can only understand that you have created and initialized a `Server` instance and added a ProjectEntry to it with the content above. The rest is unclear.

There is no problem for a test to have some more lines of code, but it is important for a test to be as clear as possible.

Reply:
Can you be more specific as to how you'd prefer it look? I'm not sure how this is unclear (or at least more unclear than tests like the Hover tests above it).

Reply:
I personally would write most tests for the simplest object - line formatter itself and add just few that involve the LS itself as to verify the integration. This helps if someone wants to borrow the formatter but doesn't need the LS - tests then are easy to take with the code.

Reply:
I think that's already how I have it set up. LineFormatterTests contains all of the test cases for the formatter itself. This test is just checking that the server does something. If you mean that I should remove any actual stylistic choices from here and do a simple check of "does the server return an edit instead of throwing NotImplemented", then I can do that too.

Reply:
I modified this test using the suggested added extension method to be more explicit, if you'd like to look again.